zstash/hpss.py

```diff
@@ -95,7 +98,28 @@ def hpss_transfer(
             # Transfer file using `hsi`
             command: str = 'hsi -q "cd {}; {} {}"'.format(hpss, transfer_command, name)
             error_str: str = "Transferring file {} HPSS: {}".format(transfer_word, name)
-            run_command(command, error_str)
+            tries = retries + 1
+            while tries > 0:
+                tries -= 1
+                try:
+                    run_command(command, error_str)
+                    # Command was successful. No need to retry.
+                    break
+                except RuntimeError as e:
+                    actual_size = os.path.getsize(name)
+                    if (
+                        (tries > 0)
+                        and (transfer_type == "get")
+                        and cur
+                        and tars_table_exists(cur)
+                    ):
+                        cur.execute(f"select size from tars where name is '{name}';")
+                        expected_size: int = cur.fetchall()[0][0]
+                        if expected_size > actual_size:
+                            # Rerun since tar was incomplete
+                            # Break out of the for-loop and run the try-except block again
+                            continue
+                    raise e
```

Comment:
@golaz @chengzhuzhang Here is the primary piece of code for this pull request. It's very challenging to test behavior that only occurs when there is an error, but I believe I have this set up correctly. Please let me know if this looks good to you as well. Thanks! (I did add a unit test, but that doesn't create an error to test the `except` block.)